src/main/java/uk/ac/ic/wlgitbridge/bridge/Bridge.java

```
@@ -289,25 +272,30 @@ public void checkDB() {
             if (f.getName().equals(".wlgb")) {
                 continue;
             }
-            String projName = f.getName();
-            try (LockGuard __ = lock.lockGuard(projName)) {
-                File dotGit = new File(f, ".git");
-                if (!dotGit.exists()) {
-                    Log.warn("Project: {} has no .git", projName);
-                    continue;
-                }
-                ProjectState state = dbStore.getProjectState(projName);
-                if (state != ProjectState.NOT_PRESENT) {
-                    continue;
-                }
-                Log.warn(
-                        "Project: {} not in swap_store, adding",
-                        projName
-                );
-                dbStore.setLastAccessedTime(
-                        projName,
-                        new Timestamp(dotGit.lastModified())
-                );
+            String projectName = f.getName();
+            try {
+                ContextStore.inContextWithLock(projectName, (context) -> {
```

I am not sure what we can do about it... but this locking operation is probably going to be quite a bit slower now, and when running in k8s I would guess we're likely to see more git bridge reboots. We should at least keep an eye on how long it takes the git bridge to boot.

If it's too slow, we might have to stop doing these checks on every boot. They don't block startup in any case (the error gets eaten below), so we would not be losing that much safety. It's possible that these checks could be done asynchronously, but I think that would require some adjustments to account for projects that might get swapped in between making the list of projects and checking the projects.